### PR TITLE
API for deleting package.

### PR DESCRIPTION
**Pull request by arteevraina**
I have created the PR for initializing the API for deleting a package. Though, I have not used the original API route that was discussed in #5 . 

I had an idea to use this route.
```
/packages/<username>/<namespace_name>/<package_name>/delete 
```
instead of
```
/<username>/<package_id>/delete
```
because I was doubtful that ``_id`` of the mongo DB document would be correctly sent by the URL.

Commits to review:
* ca1cd597453d0706ad96de9ba877bd22ce6b6f10
* fe0229c4c29bc98d9931cffea5e68b371af7d586
* fd5f614b6b45da3603c5ddd173968492aab06721

@henilp105 @minhqdao @perazz  @fortran-lang/fpm  I have few questions here, Since we have nothing merged in the main branch as of today And this PR contains changes made by @henilp105 as well since the start of project. Should I consider creating this PR to the henils remote branch and can be reviewed there as of now because it will be easier to review then.

### CODE REVIEW

**Comment by henilp105** [comment]
@arteevraina I had considered `package_name` in the field maybe it was incurred due to typo. I am considering package_name as the field in the API.

**Comment by henilp105** [comment]
@arteevraina I had considered `package_name` in the field maybe it was incurred due to typo. I am considering package_name as the field in the API.

**Comment by minhqdao** [comment]
To answer your question: The ideal case would be that PRs are quickly merged into main and new PRs are also opened and merged against main. That and keeping PR sizes small should result in a small number of merge conflicts and short development cycles.

**Comment by henilp105** [comment]
@arteevraina I have made the flask code modular which would make it easy to add new features/commits and also prevent (mostly) the merge conflicts.

File: flask/packages.py [hunk 1]
```diff
@@ -114,3 +114,32 @@ def upload():
 
         return jsonify({"message": "Package Uploaded Successfully.", "code": 200})
 
+@app.route("/packages/<username>/<namespace_name>/<package_name>/delete", methods=["GET", "POST"])
+def delete_package(username, namespace_name, package_name):
+    uuid = request.cookies.get("uuid")
+    if not uuid:
+        return render_template("login.html")
```
Comment (henilp105) [inline flask/packages.py:121]: @arteevraina We would have to add a json response of unauthorized 403, as we have shifted to react/next.js frontend and for CLI application.

File: flask/packages.py [hunk 2]
```diff
@@ -114,3 +114,32 @@ def upload():
...
+
+    user = db.users.find_one({"uuid": uuid})
+
+    # If user is null or username does not matches with the username in the route.
+    if not user or user["name"] != username:
+        return render_template("login.html")
+
+    # Find package using package_name and namespace_name.
+    package = db.packages.find_one({"name": package_name, "namespace": namespace_name})
```
Comment (henilp105) [inline flask/packages.py:130]: as we can't parse _id by url thus, We would have to find the namespace `_id` by its name, as in packages collection we have namespace's `_id` .

File: flask/packages.py [hunk 3]
```diff
@@ -114,3 +114,32 @@ def upload():
...
+
+    # If package is not found related to the package_id. Return 404.
+    if not package:
+        return jsonify({"message": "Package not found", "code": 404})
+
+    # Check if the user is authorized to delete the package.
+    if user["_id"] != package["author"] and user["_id"] not in package["maintainers"]:
```
Comment (henilp105) [inline flask/packages.py:137]: as we have considered that an author will always be a maintainer, thus we can simplify this check.

**Pull request closed without merging**